packages/fxa-payments-server/src/routes/Subscriptions/SubscriptionItem.tsx

```diff
@@ -55,6 +58,15 @@ export const SubscriptionItem = ({
     );
   }
 
+  if (
```

Comment:
I don't think this `if` is necessary. Now that we are fetching the next invoices actual amount and start date, we should always be using that data. I go into a bit more detail in the review comment below.

Reply:
so this if check is necessary because the because the subsequent invoice can be undefined (cancelled subscription)

Reply:
Are you suggesting that we add the `SubsequentInvoice` as a property on the `CancelSubscriptionPanel` component? Since `SubsequentInvoice` can be undefined, we'd have to some this if check into there. 

I'm not sure if I am following the "infers the amount" part since its just using whatever is passed in in the `plan` property. I do agree that I don't find this if check eloquent but I also don't find the display of the plan price being managed in the `CancelSubscriptionPanel` component eloquent either (I think it should probably be handled in the `SubscriptionItem` component itself). So I am generally inclined to agree with your feedback but I am not sure I'm 100% there on this one though.

This ticket was also scoped to 1 point and while this work affected this, given my stated opinion above, I am inclined to push for this to be a maintanence task (perhaps a `good-first-issue`). I'd like to avoid further delaying coupons work with scope creep if possible (and I do concede that this is debatably scope creep).

Reply:
Oh no, I was actually just thinking of adding 2 new props to the `CancelSubscriptionPanel` component, 1 for the next bill date and another for the amount you'll be billed. (In the `CancelSubscriptionPanel` component these would then replace the values `customerSubscription.current_period_end` and `plan.amount`, respectively)

My thinking behind this change is more from a concepts and re-usability perspective. At the moment, if you look just at the `CancelSubscriptionPanel` component on it's own, its basically guessing the values for next bill date and next bill amount by looking at the `Plan` data it got back from Stripe. However, since we now actually have the next bill's information available to us, via the SubsequentInvoiceAPI, my suggestion is that by adding the 2 new props it will make the `CancelSubscriptionPanel` component more clear and usable.

Reply:
Ivo filed  https://mozilla-hub.atlassian.net/browse/FXA-4704 to refactor this component